src/lib/y2partitioner/format_mount_options.rb

```diff
@@ -0,0 +1,293 @@
+    DEFAULT_HOME_FS = Y2Storage::Filesystems::Type::XFS
+
+    # Constructor
+    #
+    # @param options [Hash]
+    # @param partition [Y2Storage::BlkDevice]
+    # @param role [Symbol]
+    def initialize(options: {}, partition: nil, role: nil)
+      set_defaults!
+
+      options_for_role(role) if role
+      options_for_partition(partition) if partition
+
+      @mount = @mount_point && !@mount_point.empty?
+
+      options.each do |o, v|
+        send("#{o}=", v) if respond_to?("#{o}=")
+      end
+    end
+
+    def set_defaults!
+      @format = false
+      @encrypt = false
+      @mount_by = Y2Storage::Filesystems::MountByType::UUID
+      @filesystem_type = DEFAULT_FS
+      @partition_id = Y2Storage::PartitionId::LINUX
+      @fstab_options = []
+    end
+
+    # sets current attributes based on the given partition
+    # @param partition [Y2Storage::BlkDevice]
+    def options_for_partition(partition)
+      return unless partition
+
+      @name = partition.name
+      @partition_type = partition.type
+      @partition_id = partition.id
+
+      options_for_filesystem(partition.filesystem)
+    end
+
+    # sets current filesystem attributes based on the given one
+    # @param filesystem [Y2Storage::Filesystems::Type]
+    def options_for_filesystem(filesystem)
+      return unless filesystem
+
+      @filesystem_type = filesystem.type
+      @mount_point = filesystem.mount_point
+      @mount_by = filesystem.mount_by if filesystem.mount_by
+      @label = filesystem.label
+      @fstab_options = filesystem.fstab_options
+    end
+
+    # @param role [Symbol]
+    def options_for_role(role)
+      case role
+      when :swap
+        @mount_point = "swap"
+        @filesystem_type = Y2Storage::Filesystems::Type::SWAP
+        @partition_id = Y2Storage::PartitionId::SWAP
+        @mount_by = Y2Storage::Filesystems::MountByType::DEVICE
+      when :efi_boot
+        @mount_point = "/boot/efi"
+        @partition_id = Y2Storage::PartitionId::ESP
+        @filesystem_type = Y2Storage::Filesystems::Type::VFAT
+      when :raw
+        @partition_id = Y2Storage::PartitionId::LVM
+      else
+        @mount_point = ""
+        @filesystem = (role == :system) ? DEFAULT_FS : DEFAULT_HOME_FS
+        @partition_id = Y2Storage::PartitionId::LINUX
+      end
+    end
+
+    def options_for_windows_partition(_partition_id)
+      @filesystem_type = Y2Storage::Filesystems::Type::VFAT
+    end
+
+    # @param partition_id [Y2Storage::PartitionId]
+    def options_for_partition_id(partition_id)
+      return options_for_windows_partition(partition_id) if partition_id.is?(:windows_system)
+
+      case partition_id
+      when Y2Storage::PartitionId::SWAP
+        options_for_role(:swap)
+      when Y2Storage::PartitionId::ESP
+        options_for_role(:efi_boot)
+      else
+        @filesystem_type = partition_id.formattable? ? DEFAULT_FS : nil
+      end
+    end
+
+    def used_mount_points
```

Comment:
I will add a reminder aka FIXME: It will be used to just suggest not used mount_points.